agent/logger/log.go

```diff
@@ -14,87 +14,162 @@
 package logger
 
 import (
+	"fmt"
 	"os"
+	"strconv"
 	"strings"
 	"sync"
+	"time"
 
-	log "github.com/cihub/seelog"
+	"github.com/cihub/seelog"
 )
 
 const (
-	LOGLEVEL_ENV_VAR = "ECS_LOGLEVEL"
-	LOGFILE_ENV_VAR  = "ECS_LOGFILE"
-
-	DEFAULT_LOGLEVEL = "info"
+	LOGLEVEL_ENV_VAR           = "ECS_LOGLEVEL"
+	LOGFILE_ENV_VAR            = "ECS_LOGFILE"
+	LOG_ROLLOVER_TYPE_ENV_VAR  = "ECS_LOG_ROLLOVER_TYPE"
+	LOG_OUTPUT_FORMAT_ENV_VAR  = "ECS_LOG_OUTPUT_FORMAT"
+	LOG_MAX_FILE_SIZE_ENV_VAR  = "ECS_LOG_MAX_FILE_SIZE_MB"
+	LOG_MAX_ROLL_COUNT_ENV_VAR = "ECS_LOG_MAX_ROLL_COUNT"
+
+	DEFAULT_LOGLEVEL               = "info"
+	DEFAULT_ROLLOVER_TYPE          = "date"
+	DEFAULT_OUTPUT_FORMAT          = "logfmt"
+	DEFAULT_MAX_FILE_SIZE  float64 = 10
+	DEFAULT_MAX_ROLL_COUNT int     = 24
 )
 
-var logfile string
-var level string
-var levelLock sync.RWMutex
-var levels map[string]string
-var logger OldLogger
+type logConfig struct {
+	RolloverType  string
+	MaxRollCount  int
+	MaxFileSizeMB float64
+	logfile       string
+	level         string
+	outputFormat  string
+	lock          sync.Mutex
+}
 
-// Initialize this logger once
-var once sync.Once
+var Config *logConfig
 
-func init() {
-	once.Do(initLogger)
+func logfmtFormatter(params string) seelog.FormatterFunc {
```

Comment:
Can we add a piece of example log file in the pr description? It will be more straightforward for us to know what the log look like now.

Reply:
sure!